docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,480 @@
+
+- **scheduling**: Schedulers can often represent resources as graphs (e.g., Flux Framework), and this community would want to be able to easily map one or more compatibility schemas into their scheduler graphs.
+
+The design suggested in this proposal addresses both use cases above.
+
+### Benefits
+
+Several benefits to this design include:
+
+- Parsing (for scheduling or image selection) means we can generate one graph (and don't need to combine graphs)
+- We don't need to handle cases of "single node" graphs (metadata attributes that are nodes but don't have edges)
+- We don't need to keep around an extra jsonschema to validate a custom format, we use a common, community standard (JGF)
+- Simple cases of image selection do not need to consider the graph and can use key value pairs provided in the compatibility spec
+- Changes to the graph (nodes or edges) are in one place (one source of truth) and versioned, so any upstream change does not require updating artifacts.
+
+## Definitions
+
+- A **Compatibility Interest Group** (CIG) has a vested interest in defining compatibility.
+  - Defines namespace for their community metadata
+  - Maintains a schema "compspec.json" in a web-accessible location for their community metadata
+  - Optionally maintains plugins or associated tooling
+- A **Compatibility Schema** is a set of namespaced metadata that describes a compatibiilty interest group's space of expertise.
+  - Maintained by a compatibility interest group
+  - Metadata is described in a graph format, [Json Graph Format](https://github.com/jsongraph/json-graph-specification), version 2.0, with edges and nodes required.
+  - The graph id is the namespace or label for their community metadata (e.g,. `supercontainers.org` or `archspec.io`)
+  - Every node must be represented in the graph (defined in edges), and there should be common validation for this.
+  - Each nodes is a specific metadata attribute (e.g., `cpu` and `cpu.family`)
+  - Each edge is a relationship between two nodes (e.g., `cpu` contains -> `cpu.family`.
+- A **Compatibility Specification** is compatibility metadata that describes a container image or application
+  - Includes one or more compatibility schemas
+  - Also is validated by way of links to the compspec.json for each
+
+Importantly, there is no complex graph structure in artifact itself, but rather in the single, top level schema that defines the space of metadata attributes and relationships. The tools can use the graph (nodes and edges) as needed for their use case. This means they _are not required_ to use them and can simply view the compatibility spec (what we might refer to as an artifact) as a listing of metadata attributes. This gives power and freedom to tool developers to create tools as they see fit. Several designs are afforded:
```

Comment:
The artifact must allow to express the graph, to represent more complex compatibility relations. However, that could be an optional feature (as described [here](https://github.com/opencontainers/wg-image-compatibility/pull/6/files#diff-644786f78aa35025b6d4493a27fcc3d34c7a50df73c81c135daff93025a620c8R113)) and would not have to be present in the artifact always (up to the users).

Reply:
Anyway, it's just a requirement which we should consider. I am resolving this as it is not a change request against this proposal.